.travis.yml

```diff
@@ -6,16 +6,31 @@ services:
 language: go
 go_import_path: github.com/uber/jaeger-lib
 
-go:
-  - 1.7
+matrix:
+  include:
+  - go: 1.7
+    env:
+    - COVERAGE=true
+  - go: 1.9
+    env:
+    - TEST=true
+  - go: 1.7
+    env:
+    - TEST=true
+    - USE_DEP=true
 
 env:
   global:
     - GO15VENDOREXPERIMENT=1
 
 install:
+  - if [ "$USE_DEP" == true ]; then make install-dep-ci ; else echo 'skipping installing dep'; fi
   - make install-ci
 
 script:
-  - make test-ci
-  - travis_retry goveralls -coverprofile=cover.out -service=travis-ci || true
+  - if [ "$COVERAGE" == true ]; then make test-ci ; else echo 'skipping tests'; fi
+  - if [ "$TEST" == true ]; then make test-only-ci ; else echo 'skipping tests'; fi
+
+after_success:
+  - if [ "$COVERAGE" == true ]; then mv cover.out coverage.txt ; else echo 'skipping coverage'; fi
```

Comment:
is something preventing these 2 lines to be combined?

Reply:
yes, me knowing the syntax how to do it without running travis 5 more times